apps/req.c

```diff
@@ -765,8 +765,9 @@ int req_main(int argc, char **argv)
         }
     }
     if (newreq || gen_x509) {
-        if (pkey == NULL /* can happen only if !newreq */) {
-            BIO_printf(bio_err, "Must provide a signature key using -key\n");
+        if (CAcert == NULL && pkey == NULL /* can happen only if !newreq */) {
```

Comment:
I would actually check for NULL CAkey and not CAcert. And move the comment inside the if block or maybe even drop it altogether as it is misleading.

Reply:
Yes, these are equivalent conditions, testing for CAkey looked a little bit more readable to me however it does not really matter. So you can keep the condition as is.